### PR TITLE
Add google provider

### PR DESCRIPTION
**Pull request by eliecharra**

### CODE REVIEW

**Comment by eliecharra** [comment]
The build is failing because of 

```
Error: You are trying to create client-side redirections to paths that do not exist:
- /0.14.0/providers/google/authentication
```

This is a chicken and egg issue ...

File: docs/providers/google/authentication.mdx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---
+id: authentication
+title: Authentication
+---
+
+To use driftctl, we need credentials to make authenticated requests to your GCP project.
+Just like the terraform provider, we retrieve config from [environment variables](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_reference#full-reference).
```
Comment (wbeuil) [inline docs/providers/google/authentication.mdx:7]: `Just like the Terraform ..., we retrieve configuration from ...`

File: docs/providers/google/authentication.mdx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  CLOUDSDK_CORE_PROJECT=my-project\
+  driftctl scan --to gcp+tf
+```
+
+## Least privileged policy
+
+driftctl uses **Google Asset API** to enumerate efficiency resources on your account, and thus you need at least the role [Cloud Asset Viewer](https://cloud.google.com/iam/docs/understanding-roles#cloud-asset-roles) to be enabled.
+
+:::info deep mode
+If you want to use driftctl with deep mode, driftctl will also need to retrieve resources details and the **Cloud Asset Viewer** will not be enough.
```
Comment (wbeuil) [inline docs/providers/google/authentication.mdx:20]: `to retrieve details' resources and the ..`

File: docs/providers/google/authentication.mdx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+If you want to be able to get resources details you should setup the basic role [Viewer](https://cloud.google.com/iam/docs/understanding-roles#basic-definitions) on your project.
```
Comment (wbeuil) [inline docs/providers/google/authentication.mdx:21]: `.. to get the details you ...`

**Review by wbeuil — CHANGES_REQUESTED** [review]

**Review by wbeuil — APPROVED** [review]
LGTM, but the build failed since we do not have an actual `/0.14.0/providers/google/authentication` page.

We should wait for it to be in production before adding the redirection.

**Pull request merged**